File: Cores/JVM/src/main/scala/OSOL/Extremum/Cores/JVM/Optimization/Testing/IntervalTester.scala
package OSOL.Extremum.Cores.JVM.Optimization.Testing

import OSOL.Extremum.Cores.JVM.Arithmetics.Interval
import OSOL.Extremum.Cores.JVM.Arithmetics.Interval.Converters._
import OSOL.Extremum.Cores.JVM.Optimization.Area
import OSOL.Extremum.Cores.JVM.Optimization.RemoteFunctions.IntervalRemoteFunction
import OSOL.Extremum.Cores.JVM.Vectors.IntervalVector

class IntervalTester extends Tester[IntervalVector, Interval, IntervalVector](
  testFunctions = IntervalTester.f,
  areas = IntervalTester.a,
  solutions = IntervalTester.s,
  tolerance = IntervalTester.tolerance,
  attempts = IntervalTester.attempts) {

}

object IntervalTester {

  val tolerance = 1e-3
  val attempts = 5

  val vars_1 = Seq("x")
  val vars_2 = Seq("x", "y")
  val vars_3 = Seq("x", "y", "z")

  val TASKS_LOC = sys.env("OSOL_EXTREMUM_TASKS_LOC")

  val f1: IntervalRemoteFunction = new IntervalRemoteFunction(json = s"${TASKS_LOC}/Dummy/Dummy_1.json", port = 5000, field = "f")
  val a1: Area = vars_1.map(k => (k, (-10.0, 10.0))).toMap
  val s1: Map[String, Double] = vars_1.map(k => (k, 0.0)).toMap

  val f2: IntervalRemoteFunction = new IntervalRemoteFunction(json = s"${TASKS_LOC}/Dummy/Dummy_2.json", port = 5000, field = "f")
  val a2: Area = vars_2.map(k => (k, (-10.0, 10.0))).toMap
  val s2: Map[String, Double] = vars_2.map(k => (k, 0.0)).toMap

  val f3: IntervalRemoteFunction = new IntervalRemoteFunction(json = s"${TASKS_LOC}/Dummy/Dummy_3.json", port = 5000, field = "f")
  val a3: Area = vars_3.map(k => (k, (-10.0, 10.0))).toMap
  val s3: Map[String, Double] = vars_3.map(k => (k, 0.0)).toMap


  val f = Seq(f1, f2, f3)
  val a = Seq(a1, a2, a3)
  val s = Seq(s1, s2, s3)

}

File: Cores/DotNet/Optimization/Testing/IntervalTester.cs
using System;
using System.Collections.Generic;
using System.Linq;
using OSOL.Extremum.Cores.DotNet.Arithmetics;
using OSOL.Extremum.Cores.DotNet.Optimization.RemoteFunctions;
using OSOL.Extremum.Cores.DotNet.Vectors;

namespace OSOL.Extremum.Cores.DotNet.Optimization.Testing
{
    public class IntervalTester: Tester<IntervalVector, Interval, IntervalVector>
    {
        
        public static string TASKS_LOC = Environment.GetEnvironmentVariable("OSOL_EXTREMUM_TASKS_LOC");

        public IntervalTester()
        {
            this.Tolerance = 1e-3;
            this.Attempts = 5;

            var vars_1 = new[] {"x"};
            var vars_2 = new[] {"x", "y"};
            var vars_3 = new[] {"x", "y", "z"};
            
            RemoteFunction<Interval> f1 = new IntervalRemoteFunction(json: $"{TASKS_LOC}/Dummy/Dummy_1.json", port: 11121, field: "f");
            Dictionary<string, Tuple<double, double>> a1 = vars_1.ToDictionary(k => k, k => Tuple.Create(-10.0, 10.0));
            Dictionary<string, double> s1 = vars_1.ToDictionary(k => k, k => 0.0);
            
            RemoteFunction<Interval> f2 = new IntervalRemoteFunction(json: $"{TASKS_LOC}/Dummy/Dummy_2.json", port: 11122, field: "f");
            Dictionary<string, Tuple<double, double>> a2 = vars_2.ToDictionary(k => k, k => Tuple.Create(-10.0, 10.0));
            Dictionary<string, double> s2 = vars_2.ToDictionary(k => k, k => 0.0);
            
            RemoteFunction<Interval> f3 = new IntervalRemoteFunction(json: $"{TASKS_LOC}/Dummy/Dummy_3.json", port: 11123, field: "f");
            Dictionary<string, Tuple<double, double>> a3 = vars_3.ToDictionary(k => k, k => Tuple.Create(-10.0, 10.0));
            Dictionary<string, double> s3 = vars_3.ToDictionary(k => k, k => 0.0);

            this.TestFunctions = new[] {f1, f2, f3};
            this.Areas = new[] {a1, a2, a3};
            this.Solutions = new[] {s1, s2, s3};

        }
        
    }
}